medusa/storage/google_cloud_storage/gsutil.py

```diff
@@ -88,10 +88,12 @@ def cp(self, *, srcs, dst, max_retries=5, parallel_process_count=1):
                 '-o', '\"GSUtil:parallel_process_count={}\"'.format(parallel_process_count),
             ]
 
-        if self._config.storage_class is None:
-            storage_class_options = []
-        else:
-            storage_class_options = ["-s", self._config.storage_class]
+        storage_class_options = []
+
+        # storage class option works with cp upload only
+        if dst.startswith('gs://'):
```

Comment:
This is incorrect as you can see [here](https://github.com/thelastpickle/cassandra-medusa/actions/runs/4594351406/jobs/8113516504#step:8:251). dst is a PosixPath object which doesn't have a `startsWith()` method.

Please make sure you've tested the changes that are being pushed otherwise it'll create inertia in the review process. I understand this can be challenging since there are several cloud providers here.
Sadly we cannot enable cloud testing for externally contributed PRs due to how secrets are managed in GitHub Actions (which is no different than other CI systems btw, for security reasons).

Reply:
Hi @adejanovski 

Sorry for that. 
I do not have a GCP environment to test my changes. I'll create a test environment soon for my next PRs. 
I've fixed the issue by converting the Posixpath to str and I tested my changes. 

Thanks